opacus/data_loader.py

```diff
@@ -143,13 +144,15 @@ def __init__(
         if collate_fn is None:
             collate_fn = default_collate
 
+        if drop_last:
+            warnings.warn("Ignoring drop_last as it is not compatible with DPDataLoader.")
```

Comment:
nit: As per @romovpa's comment on #352, warning is probably not necessary as the client doesn't have to take any action. How about changing this to logger.warning() instead?